File: 2-add-two-numbers/add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        int carry = 0;
        ListNode prev = null;
        ListNode fn = l1, sn = l2;
        while(fn != null || sn != null) {
            int fv = fn != null ? fn.val : 0;
            int sv = sn != null ? sn.val : 0;
            int value = (fv + sv + carry) % 10;
            carry = (fv + sv + carry) / 10;
            if(sn == null) {
                ListNode node = new ListNode(value);
                prev.next = node;
                prev = node;
            } else {
                sn.val = value;
                prev = sn;
                sn = sn.next;
            }
            if(fn != null) {
                fn = fn.next;
            }
        }
        if(carry != 0) {
            ListNode node = new ListNode(carry);
            prev.next = node;
        }
        return l2;
    }
}

File: 2-add-two-numbers/add-two-numbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def addTwoNumbers(self, l1, l2):
        carry = 0
        prev = ListNode(-1)
        pointer = prev
        fn,sn = l1,l2
        while fn is not None or sn is not None or carry != 0:
            fv = fn.val if fn is not None else 0
            sv = sn.val if sn is not None else 0
            value = (fv + sv + carry) % 10
            carry = (fv + sv + carry) / 10
            new_node = ListNode(value)
            pointer.next = new_node
            pointer = pointer.next
            fn = fn.next if fn is not None else None
            sn = sn.next if sn is not None else None
        
        return prev.next
        